can't build static lib

**Describe the bug**
I want to build a static lib. so I set :
set(BUILD_SHARED_LIBS OFF)
set(BUILD_STATIC_LIBS ON)
in CMakeList.txt
But I can't make it:

Scanning dependencies of target lz4_static
[  6%] Building C object CMakeFiles/lz4_static.dir/home/smit/resource_home/lz4-dev/lib/lz4.c.o
[ 13%] Building C object CMakeFiles/lz4_static.dir/home/smit/resource_home/lz4-dev/lib/lz4hc.c.o
[ 20%] Building C object CMakeFiles/lz4_static.dir/home/smit/resource_home/lz4-dev/lib/lz4frame.c.o
[ 26%] Building C object CMakeFiles/lz4_static.dir/home/smit/resource_home/lz4-dev/lib/xxhash.c.o
[ 33%] Linking C static library liblz4.a
[ 33%] Built target lz4_static
Scanning dependencies of target lz4c
[ 40%] Linking C executable lz4c
CMakeFiles/lz4c.dir/home/smit/resource_home/lz4-dev/programs/bench.c.o: In function `BMK_benchMem':
bench.c:(.text+0x193d): undefined reference to `XXH64'
bench.c:(.text+0x24e1): undefined reference to `XXH64'
collect2: error: ld returned 1 exit status
make[2]: *** [lz4c] Error 1
make[1]: *** [CMakeFiles/lz4c.dir/all] Error 2
make: *** [all] Error 2

**To Reproduce**
Steps to reproduce the behavior:
1. Go to  'build/cmake/CMakeList.txt'
2. add set(BUILD_SHARED_LIBS OFF)  set(BUILD_STATIC_LIBS ON) On line 66
3. cmake
4. make

make error

**System (please complete the following information):**
 - OS: Centos 7.9
 - Version 7.9
 - Compiler gcc 9.4
 - Build System cmake
 - Other hardware specs [e.g. Core 2 duo...]

--- a/build/cmake/CMakeLists.txt
+++ b/build/cmake/CMakeLists.txt
@@ -86,6 +86,7 @@ set(LZ4_CLI_SOURCES
   "${LZ4_PROG_SOURCE_DIR}/lz4cli.c"
   "${LZ4_PROG_SOURCE_DIR}/lz4io.c"
   "${LZ4_PROG_SOURCE_DIR}/datagen.c")
+list(APPEND LZ4_CLI_SOURCES ${LZ4_SOURCES}) # LZ4_CLI always use liblz4 sources directly.
 
 # Whether to use position independent code for the static library.  If
 # we're building a shared library this is ignored and PIC is always
@@ -135,20 +136,11 @@ if(BUILD_STATIC_LIBS)
     XXH_NAMESPACE=LZ4_)
 endif()
 
-if(BUILD_STATIC_LIBS)
-  set(LZ4_LINK_LIBRARY lz4_static)
-else()
-  list(APPEND LZ4_CLI_SOURCES ${LZ4_SOURCES})
-endif()
-
 # lz4
 if (LZ4_BUILD_CLI)
   set(LZ4_PROGRAMS_BUILT lz4cli)
   add_executable(lz4cli ${LZ4_CLI_SOURCES})
   set_target_properties(lz4cli PROPERTIES OUTPUT_NAME lz4)
-  if (BUILD_STATIC_LIBS)
-    target_link_libraries(lz4cli ${LZ4_LINK_LIBRARY})
-  endif()
 endif()
 
 # lz4c
@@ -156,9 +148,6 @@ if (LZ4_BUILD_LEGACY_LZ4C)
   list(APPEND LZ4_PROGRAMS_BUILT lz4c)
   add_executable(lz4c ${LZ4_CLI_SOURCES})
   set_target_properties(lz4c PROPERTIES COMPILE_DEFINITIONS "ENABLE_LZ4C_LEGACY_OPTIONS")
-  if (BUILD_STATIC_LIBS)
-    target_link_libraries(lz4c ${LZ4_LINK_LIBRARY})
-  endif()
 endif()
 
 # Extra warning flags

